### PR TITLE
Add ability to fetch for Salesforce Object meta data to ForceAPI

### PR DESCRIPTION
**Pull request by k4jiang**
Reference: https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/resources_sobject_describe.htm

Exposes endpoint that describes the individual metadata at all levels for a specified Salesforce object.

### CODE REVIEW

File: src/main/java/com/springml/salesforce/wave/api/ForceAPI.java [hunk 1]
```diff
@@ -47,4 +48,12 @@
...
 
     public String getSFEndpoint() throws Exception;
+
+    /**
+     * Returns list of the salesforce object column names
+     * @param object Name of the salesforce object
+     * @return
+     * @throws Exception
+     */
+    public DescribeSObjectResult describeSalesforceObject(String object) throws Exception;
```
Comment (mengxr) [inline src/main/java/com/springml/salesforce/wave/api/ForceAPI.java]: similar compatibility issue here, we need to confirm spark-salesforce is the only downstream app. @samuel-pt

File: src/main/java/com/springml/salesforce/wave/model/DescribeSObjectResult.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
+package com.springml.salesforce.wave.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import java.io.Serializable;
+import java.util.List;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class DescribeSObjectResult implements Serializable {
```
Comment (mengxr) [inline src/main/java/com/springml/salesforce/wave/model/DescribeSObjectResult.java]: just curious, is this bean class generated or hand-written?

Comment (k4jiang) [reply src/main/java/com/springml/salesforce/wave/model/DescribeSObjectResult.java]: The private properties were hand-written. The getters and setters were generated.

File: src/main/java/com/springml/salesforce/wave/api/ForceAPI.java (hunk 1 above)
Comment (samuel-pt) [reply src/main/java/com/springml/salesforce/wave/api/ForceAPI.java]: @mengxr  Currently spark-salesforce is the only downstream app. So no issues

**Pull request merged**